index.ts

```diff
@@ -55,21 +56,22 @@ export class SpotifyClient {
       })
     });
     const responseData = (await response.json()) as AccessToken;
-    return responseData?.access_token;
+    return responseData;
   };
 
   getCurrentlyPlaying = async ({
     fallbackToLastPlayed = false
   }: CurrentlyPlayingOptions = {}) => {
     try {
-      const accessToken = await this._genAccesToken();
-      const headers = { Authorization: `Bearer ${accessToken}` };
+      if(this.accessToken === null) 
+        this.accessToken = await this._genAccesToken();
+      const headers = { Authorization: `Bearer ${this.accessToken.access_token}` };
       const response = await fetch(CURRENTLY_PLAYING_URL, { headers });
       let isPlaying = true;
       if (response.status === 204) {
         isPlaying = false;
         return fallbackToLastPlayed
-          ? { isPlaying, ...(await this.getLastPlayed()) }
+          ? { isPlaying, ...(await this.getLastPlayed())[0] as ResponseTrack }
```

Comment:
the type here is already inferred with the return type of `getLastPlayed`. I don't need to think we need explicitly add typing here

Reply:
you're right. this is not necessary and should be removed